### PR TITLE
Added user-agent to visits

### PR DESCRIPTION
**Pull request by jrasanen**

### CODE REVIEW

**Comment by parkr** [comment]
Hey @jrasanen! Thanks for your pull request. Can you help me understand why this is something worthwhile keeping track of?

**Comment by jrasanen** [comment]
@parkr My use case needs user agent info for querying browser statistics. What do you think about the decision to return an [error if browser](https://github.com/parkr/ping/pull/3/files#diff-1ce7d97ddd43423f168d3cad79898551R92) does not provide a user agent?

**Comment by jrasanen** [comment]
I also understand if this is in conflict with the original purpose to just collect most popular pages, so no worries if you close this PR. :)

File: ping.go [hunk 1]
```diff
@@ -84,10 +84,20 @@ func ping(w http.ResponseWriter, r *http.Request) {
...
 	}
 
+	var ua string
+	if res := r.Header.Get("User-Agent"); res != "" {
+		ua = res
+	} else {
+		log.Println("empty user-agent")
+		javascriptRespond(w, http.StatusBadRequest, "empty user-agent")
+		return
+	}
```
Comment (parkr) [inline ping.go]: I think I would refactor this into:

```go
userAgent := r.Header.Get("User-Agent")
if userAgent == "" {
    log.Println("empty user-agent")
    javascriptRespond(w, http.StatusBadRequest, "empty user-agent")
    return
}
```

I think it's a bit more direct that way. What do you think?

**Comment by parkr** [comment]
@jrasanen No problem! I think it's a good addition. One code comment above.

What do you think about generalizing this so that the user agent isn't so specific? Maybe optsions could be: `Chrome`, `Firefox`, `Safari`, `Internet Explorer`, `Other Webkit`, `utility` (e.g. `curl`/`wget`), `bot` (e.g. `googlebot`)?

File: ping.go (hunk 1 above)
Comment (jrasanen) [inline ping.go]: Thanks! Good catch.

**Comment by parkr** [comment]
Great! Thank you. 😄

**Comment by jrasanen** [comment]
@parkr Your point about generalizing the UA, it would be discarding pretty good information (Mobile device vs Desktop vs CLI utility). But it certainly could be structured better than just the plain ua string.

**Pull request merged**